AK/Assertions.cpp

```diff
@@ -5,14 +5,70 @@
  */
 
 #include <AK/Assertions.h>
-#include <AK/Format.h>
+#include <AK/Platform.h>
+#include <AK/StringView.h>
+#include <stdio.h>
+#include <string.h>
+
+#if defined(AK_OS_LINUX) || (defined(AK_OS_BSD_GENERIC) && !defined(AK_OS_OPENBSD))
```

Comment:
It seems that`<execinfo.h>` isn't a thing on Windows, MinGW, Android or Solaris, but *is* a thing on desktop Linux, macOS, and other BSDs.

@nipos From what I can tell, on OpenBSD, there's a popular libexeinfo package that would require linking it in if we want the pretty backtrace going to the console in all areas? I'm going to go with this `#ifdef` for calling `backtrace(3)` until proven otherwise :)

Reply:
OpenBSD also supports execinfo, see the execinfo.h in their source: https://github.com/openbsd/src/blob/372711c823223cae677ea19701648dc28f8874f1/gnu/lib/libexecinfo/execinfo.h
Looks like they've taken their implementation from NetBSD.

Additionally, you can add Solaris to the list of supported systems.
Their execinfo.h dates back to when Sun Microsystems was still a thing: https://github.com/illumos/illumos-gate/blob/master/usr/src/head/execinfo.h

Haiku also supports execinfo.h.
I can't find it in the Git repository unfortunately, but the file exists on my running Haiku installation and the header looks like the implementation is taken from FreeBSD, so should be compatible with the others.

Reply:
Sounds good. I slapped a `-lexecinfo` for *BSD, but perhaps that's not 100% correct. Added Solaris to the check as well. This feature should make it a lot easier to quickly debug crashes :)